packages/auth/src/providers/cognito/credentialsProvider/credentialsProvider.ts

```
@@ -50,13 +54,20 @@ export class CognitoAWSCredentialsAndIdentityIdProvider
 		const isAuthenticated = getCredentialsOptions.authenticated;
 		const tokens = getCredentialsOptions.tokens;
 		// TODO: refactor use the this._authConfig
-		const authConfig =
-			getCredentialsOptions.authConfig as UserPoolConfigAndIdentityPoolConfig;
+		const authConfig = getCredentialsOptions.authConfig;
+
+		try {
+			assertIdentityPooIdConfig(authConfig?.Cognito);
+		} catch (_err) {
+			// No identity pool configured, skipping
+			return;
+		}
```

I wonder if this code is even needed. What should be the purpose of it ? What would be the side effect of swallowing the error ?

---

If you don't have identityPool configured, could be a valid use case if you have configured UserPools only.